bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -821,34 +848,36 @@ protected void release(boolean wait, boolean writer)
       final int refCount = this.refCount.decrementAndGet();
       if (canEvict(refCount, writer)) {
         assert (next != null);
-        final Runnable storer = getStorer(data, readingOffset, writingOffset, storage);
-        if (future != null && future.cancel(false)) {
-          logger.debug("Block {} future is cancelled", this);
-        }
-        final int numberOfInMemBlockPermits = DataList.this.numberOfInMemBlockPermits.get();
-        if (wait && numberOfInMemBlockPermits == 0) {
-          future = null;
-          storer.run();
-        } else if (numberOfInMemBlockPermits < MAX_COUNT_OF_INMEM_BLOCKS / 2) {
-          future = storageExecutor.submit(storer);
+        if (storage != null) {
```

Comment:
I agree that a comprehensive check should be done by the publisher as between this block release and publisher resuming, another late subscriber may acquire block far behind from a subscriber. At the same time, till late subscriber catches up with other subscribers, enabling publisher will lead to the publisher being immediately disabled, leading to unnecessary enable/disable sequence. It should be possible to count the number of blocks till the last one and enable publisher if it is less than the max.